scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpression.scala

```diff
@@ -55,7 +55,7 @@ final case class ParsedSpelExpression(
   @volatile var parsed: Expression = initial
 
   def getValue[T](context: EvaluationContext, desiredResultType: Class[_]): T = {
```

Comment:
Maybe getValue should be synchronized?

Reply:
It can be but doesn't have to; It's Spel's internal handling of expression in `parsed.getValue` that seems to be problematic and this part has to by synchronized

Reply:
After consulting the problem with @arkadius he found the root issue with how spel might have problems during concurrent evaluation with argument conversions and we modified it accordingly